### DISCUSSION TITLE
Track_mouse_position

### DISCUSSION THREAD

**Discussion by HametAl** [pyvista/pyvista#4154] [Q&A]
I need to track mouse position (not track_click_position), but when it returns "None". Here is the example:

```python

plotter = pyvistaqt.BackgroundPlotter()
Print(plotter.track_mouse_position())
```

**Comment by HametAl** [comment]
Update:

```python
plotter = pyvistaqt.BackgroundPlotter()
plotter.track_mouse_position()
Print(plotter.pick_mouse_position())
```

This code is working now. I suggested you add this in the documentations that track_mouse_position() and pick_mouse_position()) should be used together.

Another problem I still have is that on the mouseMoveEvent plotter doesn't get updated. I mean neither the plotter nor the status bar. I want to draw a line, while the first point is fixed and the second point moves with mouse movement. Also, I want to show the mose position in the status bar.

**Comment by banesullivan** [comment]
We do not document this at all -- which needs to be fixed, but the constantly updates and stores the mouse position in the `self.mouse_position` attribute.

Usage looks like:

```py
plotter = pyvistaqt.BackgroundPlotter()

# Tracking is enabled
plotter.track_mouse_position()
...
# Fetch the last known mouse position
print(plotter.mouse_position)
```

-----

In the weeds: `track_mouse_position()` is a thin wrapper around `store_mouse_position()` to configure it as a callback on mouse move events
